src/SafeReadIOCallback.cpp

```diff
@@ -22,28 +22,28 @@ SafeReadIOCallback::EndOfStreamX::EndOfStreamX(std::size_t MissingBytes)
 
 // ----------------------------------------------------------------------
 
-SafeReadIOCallback::SafeReadIOCallback(std::shared_ptr<IOCallback> & IO) {
-  Init(IO);
+SafeReadIOCallback::SafeReadIOCallback(std::unique_ptr<IOCallback> IO) {
+  Init(std::move(IO));
 }
 
 SafeReadIOCallback::SafeReadIOCallback(void const *Mem,
                                        std::size_t Size) {
-  std::shared_ptr<IOCallback> t = std::make_shared<MemReadIOCallback>(Mem, Size);
-  Init(t);
+  auto t = std::make_unique<MemReadIOCallback>(Mem, Size);
+  Init(std::move(t));
```

Comment:
Why not change this into a single line?

```
  Init(std::make_unique<…>(…));
```

Reply:
Could work. In a different project, I had to do it this way as the function took a const shared_ptr by ref. GCC’s -fanalyzer reported a memory leak in such a case.